spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/integration/IntegrationAutoConfiguration.java

```diff
@@ -131,4 +138,23 @@ public IntegrationMBeanExporter integrationMbeanExporter() {
 
 	}
 
+	/**
+	 * Integration JDBC configuration.
+	 */
+	@Configuration
+	@ConditionalOnClass(JdbcMessageStore.class)
+	@ConditionalOnBean(DataSource.class)
```

Comment:
OK. In the light of this suggestion should we also change conditions in [`BatchAutoConfiguration`](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/batch/BatchAutoConfiguration.java#L83) and [`JdbcSessionConfiguration`](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/JdbcSessionConfiguration.java#L43)?